uint8[] ids
bool[] directions
int32[] speeds
